packages/fiori/src/ShellBar.ts

```diff
@@ -939,12 +1004,12 @@ class ShellBar extends UI5Element {
 				press: this._handleSearchIconPress.bind(this),
 				show: !!this.searchField.length,
 			},
-			...this.items.map((item, index) => {
-				item._getRealDomRef = () => this.getDomRef().querySelector(`*[data-ui5-stable=${item.stableDomRef}]`);
+			...this.items.map((item: ShellBarItem) => {
+				item._getRealDomRef = () => this.getDomRef()!.querySelector(`*[data-ui5-stable=${item.stableDomRef}]`)!;
 				return {
 					icon: item.icon,
 					id: item._id,
-					count: item.count || undefined,
+					count: item.count as unknown as number || undefined,
```

Comment:
since count may contain value like 99+ saying that its type is number seems to be not correct. i would prefer to change it to string.